### PR TITLE
feat: include only output dir

### PR DESCRIPTION
**Pull request by andrewvasilchuk**
I want to add `vue-svg-loader` to my project, but it conflicts with your created rule in `module.js`:
```js
config.module.rules.push({
  test: /\.svg$/,
  use: [
    {
      loader: 'url-loader',
      options: {
        limit: 12, <-- what a magic constant?
      },
    },
  ],
})
```
 So... I included only folder where generated sprite is stored. Also notice that this line of code

```js
const urlLoader = config.module.rules.find(rule => String(rule.test).includes('svg'))
```

does not get actual rule with `url-loader`, what it does – it gets the  first rule that match `.svg` files (in my case, my defined svg-rule with `vue-svg-loader`).  Additionally, can you, please, answer some of my question? 
1. Why do you need `url-loader` for svg-sprite file if it fallbacks to `file-loader`?
2. You have specified a `limit` in **12** bytes – it simply will never match. What's the 12 bytes mean?

### CODE REVIEW

**Comment by farnabaz** [comment]
> Why do you need url-loader for svg-sprite file if it fallbacks to file-loader?

`file-loader` increase js chunk, it places base64 encoded file in js chunk, that is not aligned with module goal.

> You have specified a limit in 12 bytes – it simply will never match. What's the 12 bytes mean?

This limit is small enough in order to don't match any file, and keep sprite files out of js chunk


I thought we could do a little improvement in extending build, to remove further conflicts with any rule.  
Exclude `gen` directory from default `svg` loader

**Comment by andrewvasilchuk** [comment]
@farnabaz , We could iterate over all rules that matches `.svg` files, and push to [`exclude`](https://webpack.js.org/configuration/module/#condition) property `gen` folder. What do you think about it?

**Comment by farnabaz** [comment]
@andrewvasilchuk
It seems good solution, `gen` directory belongs to our module only and this way we can prevent conflicts.

**Comment by andrewvasilchuk** [comment]
The `file-loader` resolves `import/require()` on a file into a url and emits the file into the output directory.

`ulr-loader` – a loader for webpack which transforms files into base64 URIs.

Seems like we can drop `url-loader`?

**Comment by farnabaz** [comment]
> Seems like we can drop `url-loader`?

You are right, we could use `file-loader` for sprites

File: lib/module.js [hunk 1]
```diff
@@ -88,12 +88,36 @@ function setupHooks(options) {
   })
 }
 
-function extendBuild(config) {
-  const urlLoader = config.module.rules.find(rule => String(rule.test).includes('svg'))
+function extendBuild(config, options) {
+  const urlLoader = config.module.rules.find(rule => {
```
Comment (farnabaz) [inline lib/module.js]: I think we don't need to find `svg` rule, all we need is excluding output directory from all rules

```js
// exclude output directory from all the loaders
config.module.rules.forEach(rule => {
  if (!rule.exclude) rule.exclude = []
  rule.exclude.push(path.resolve(options.output))
})
```

And add our custom rule
```js
config.module.rules.push({
    test: /\.svg$/,
    include: [
      path.resolve(options.output)
    ],
    use: [{
      loader: 'file-loader'
    }]
  })
```

Comment (farnabaz) [reply lib/module.js]: also please remove `package-lock.json` from PR

Comment (andrewvasilchuk) [reply lib/module.js]: How do I include your changes into my PR?

**Comment by andrewvasilchuk** [comment]
Can I create a new PullRequest?

**Pull request closed without merging**